### PR TITLE
Web Component support

### PR DESCRIPTION
**Pull request by robbiepitts**
This adds support for Web Components but does not add one to the index.html.

Relies on https://github.com/glimmerjs/glimmer-application-pipeline/pull/25

- [x] WC support
- [x] Optional WC support
- [ ] Bump @glimmer/application-pipeline

### CODE REVIEW

File: files/.babelrc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "presets": [
+    [
+      "env",
```
Comment (robbiepitts) [inline files/.babelrc]: babel-preset-env is more flexible than babel-preset-es2015 and I'm thinking probably a good fit as the default for @glimmer/application-pipeline.

File: files/.babelrc [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      {
+        "exclude": [
+          "transform-es2015-classes"
```
Comment (robbiepitts) [inline files/.babelrc]: Necessary for Web Components

File: files/src/index.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import App from './main';
 import { ComponentManager, setPropertyDidChange } from '@glimmer/component';
+import initializeCustomElements from '@glimmer/web-component';
+import customElementDefinitions from './config/custom-element-definitions';
```
Comment (tomdale) [inline files/src/index.ts]: Can we eliminate the config file and just make it an array of component names you pass?

File: files/src/index.ts [hunk 4]
```diff
@@ -17,3 +19,5 @@ app.registerInitializer({
 app.boot();
 
 app.renderComponent('<%= component %>', containerElement, null);
+
+initializeCustomElements(app, customElementDefinitions);
```
Comment (tomdale) [inline files/src/index.ts]: so like
```js
initializeCustomElements(app, ['<%= component %>', containerElement);
```

File: files/src/index.ts (hunk 3 above)
Comment (robbiepitts) [inline files/src/index.ts]: ✅

**Pull request merged**